presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergSplitSource.java

```diff
@@ -91,6 +91,8 @@ public void close()
     {
         try {
             closer.close();
+            this.fileScanTaskIterable = CloseableIterable.empty();
+            this.fileScanTaskIterator = CloseableIterator.empty();
```

Comment:
@wypb I saw fileScanTaskIterator  was already registered in the constructor: `closer.register(fileScanTaskIterator);` and `closer.close();` was supposed to close it already. Do you know why it has to be set as `CloseableIterator.empty()`? Btw the pictures you attached were not viewable..

Reply:
I debugged the `closer.close()` of the `ebergSplitSource` class and found that for the `fileScanTaskIterable` (actually an instance of the `org.apache.iceberg.util.ParallelIterable` class) object, the close method cannot release the data held by the `ParallelIterable.iterables`, which contains `org.apache.iceberg.DeleteFileIndex` object.